import-scripts/study_directory_uses_updated_filenames.py

```diff
@@ -0,0 +1,538 @@
+            sys.exit(1)
+        split_wildcard_pattern = {"prefix" : pattern_segments[0], "suffix" : pattern_segments[1]}
+        split_wildcard_pattern_list.append(split_wildcard_pattern)
+    return split_wildcard_pattern_list
+
+def matches_wildcard_pattern(filename, split_wildcard_pattern):
+    """
+    returns True if filename starts with the prefix from, and ends with the suffix from, the split_wildcard_pattern dictionary.
+    """
+    return filename.startswith(split_wildcard_pattern["prefix"]) and filename.endswith(split_wildcard_pattern["suffix"])
+
+def matches_any_wildcard_pattern(filename, split_wildcard_patterns):
+    """
+    iterates through all pattern dictionaries, looking for any match for filename.
+    """
+    for split_wildcard_pattern in split_wildcard_patterns:
+        if (matches_wildcard_pattern(filename, split_wildcard_pattern)):
+            return True
+    return False
+
+def study_directory_uses_updated_filenames(path, cancer_study_identifier=None):
+    """
+    determines if unsupported filename patterns (those added June 2021) are present in a study directory
+
+    search directory at <path> - return true if any filenames are present, which:
+        - are not of the original set of filenames we recognized before the renaming effort, and
+        - would be imported according to the current datatypes sheet, and
+        - are not "normal sample" files (which are ignored by the importer)
+
+    arguments:
+    path -- the (absolute or relative) filename path to the directory which should be examined
+    cancer_study_identifier -- cancer study stable id [optional]. If not provided, the value is searched for in meta files within <path>
+    """
+    new_filenames_present = False
+    if os.path.isdir(path):
+        path = path + os.path.sep
+    basedir = os.path.dirname(path)
+    filenames = os.listdir(path)
+    if cancer_study_identifier is None:
+        cancer_study_identifier = determine_cancer_study_identifier(basedir, filenames)
+    if cancer_study_identifier is None:
+        sys.stderr.write("Error : unable to determine cancer study identifier from meta files in study directory : '" + basedir + "'\n")
+        sys.exit(1)
+    original_datatype_filenames, original_wildcard_patterns = filenames_in_use_before_update(cancer_study_identifier)
+    current_datatype_filenames, current_wildcard_patterns = parse_current_datatypes(cancer_study_identifier)
+    split_original_wildcard_patterns = split_wildcard_patterns(original_wildcard_patterns)
+    split_current_wildcard_patterns = split_wildcard_patterns(current_wildcard_patterns)
+    for filename in filenames:
+        filepath = os.path.join(basedir, filename)
+        if not os.path.isfile(filepath):
+            # ignore subdirectories
+            continue
+        if filename in original_datatype_filenames or matches_any_wildcard_pattern(filename, split_original_wildcard_patterns):
```

Comment:
could this and the check below be put in the same line?

Reply:
or I guess since they are checking for separate specific things it's good to be separate nvm

Reply:
yes .. probably clearer to have them separate